### DISCUSSION TITLE
How can I pass an env variable to a Next.JS app running inside a Docker container?

### DISCUSSION THREAD

**Discussion by henriquedevelops** [vercel/next.js#48929] [Help]
### Summary

I am trying to pass the server URL as an env variable to a Next.JS app running inside a Docker container

[Next.JS documentation](https://nextjs.org/docs/basic-features/environment-variables) states that: "Next.js has built-in support for loading environment variables from .env.local into process.env." and "In order to expose a variable to the browser you have to prefix the variable with NEXT_PUBLIC_."

So I created an env file called ".env.local" in the root folder of the project (same folder as the docker-compose.yaml file) with the content: 
```
NEXT_PUBLIC_SERVER_URL=http://localhost:80
```

I updated the docker-compose.yaml file like this:
```
version: "3.8"
services:

  client:
    build: ./client
    ports:
      - "3000:3000"
    env_file:
      - .env.local
```

However I am unable to read the client env file in the code.

I am trying to retrieve it like Next.JS documentation suggests, I guess:

```
import axios from "axios";

export default axios.create({
  baseURL: `${process.env.NEXT_PUBLIC_SERVER_URL}/api/v1/`,
});
```

I also tried:
```
const serverUrl = process.env.NEXT_PUBLIC_SERVER_URL;

export default axios.create({
  baseURL: `${serverUrl}/api/v1/`,
});

```

It returns undefined

**Comment by icyJoseph** [comment]
I guess this means that the `.env` file is not available when the `next start` command is issued. Could you verify this by `docker run -it <image-name> sh` I think that let's you explore the image's file system, verify that `.env.local` is where it should be.

**Reply by henriquedevelops** [reply]
It is not, there is no .env.local file

**Comment by JasperAlexander** [comment]
Have you tried using `./.env.local`?
